File: data/EXP01/python/2266/2266.output_2.py
# @lc app=leetcode id=2266 lang=python3
class Solution:
    def countTexts(self, pressedKeys: str) -> int:
        mod = 1000000007
        dp = [0] * (len(pressedKeys) + 1)
        dp[0] = 1
        for i in range(1, len(pressedKeys) + 1):
            for j in range(i - 1, -1, -1):
                if pressedKeys[j:i] in {"2", "3", "4", "5", "6", "7", "8", "9"}:
                    dp[i] = (dp[i] + dp[j]) % mod
        return dp[-1]

File: data/EXP01/golang_no_system_var/2266/2266.output_2.go
func countTexts(pressedKeys string) int {
    letters := []string{"", "", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"}
    mod := int(math.Pow10(9)) + 7

    var generateMessages func(idx int, currentMessage string, digitCount []int, letters []string, totalMessages int) int
    generateMessages = func(idx int, currentMessage string, digitCount []int, letters []string, totalMessages int) int {
        if idx == len(pressedKeys) {
            return (totalMessages + 1) % mod
        }

        digit := int(pressedKeys[idx] - '0')
        for _, letter := range letters[digit] {
            digitCount[digit]++
            totalMessages = generateMessages(idx+1, currentMessage+string(letter), digitCount, letters, totalMessages)
            digitCount[digit]--
        }

        return totalMessages
    }

    digitCount := make([]int, 10)
    totalMessages := generateMessages(0, "", digitCount, letters, 0)

    return totalMessages
}
